packages/modals/src/elements/DrawerModal/Header.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import { useModalContext } from '../../utils/useModalContext';
+import { StyledDrawerModalHeader } from '../../styled';
+
+/**
+ * Accepts all `<div>` props
+ */
+export const Header = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
+  (props, ref) => {
+    const { getTitleProps } = useModalContext();
+
```

Comment:
Like we chatted about, in lieu of a possibly more comprehensive fix for the close button functionality in Modals, maybe [we follow Tooltip Modal's lead](https://github.com/zendeskgarden/react-components/blob/main/packages/modals/src/styled/StyledTooltipModalTitle.ts#L19) of padding the Header container so the content doesn't [run under the absolutely positioned Close button.](https://share.getcloudapp.com/wbu72y4Z)

Reply:
Unfortunately, `TooltipModal` is designed differently enough that `DrawerModal` can't have similar styles here. 

`TooltipModal` has padding on the dialog element and no horizontal rules. The `TooltipModal` header padding would reduce each other sides to `0px`, and apply them instead to the dialog which would then put spacing at the start and end of the horizontal rules. The end padding could be adjusted to, say `60px`, to compensate for the `Close` button. 

If this is the path chosen, `Modal`'s header should be updated since this was mostly a copy 🍝 job based on what's in production.

Reply:
I would defer to y'all on the correct approach holistically but I think it's pretty important the title content doesn't overlap the close button.

Reply:
Updated the padding to compensate spacing for the close icon:

**BEFORE:**
<img width="1073" alt="Screen Shot 2020-09-29 at 8 57 46 AM" src="https://user-images.githubusercontent.com/1811365/94583869-cf089a80-0232-11eb-8a07-6ad6d781b3fd.png">

**AFTER:**
<img width="1073" alt="Screen Shot 2020-09-29 at 8 58 21 AM" src="https://user-images.githubusercontent.com/1811365/94583850-cadc7d00-0232-11eb-95ec-6fc4bc7290c8.png">

Reply:
@hzhu did you consider the conversation under https://github.com/zendeskgarden/react-components/pull/835? All elements of the modal are optional. In this case, removing the `Close` would result in a pretty severe visual imbalance for the `Header` title.

Reply:
In that previous PR, I think we came to the conclusion that overflow prevention was left as an exercise for the component consumer (i.e. add a fixed-width `span`), depending on modal elements included + string length.

Reply:
I had nearly forgotten a PR was opened for this. That sounds fair, `DrawerModal` should align with `Modal`. I'm going to revert this particular commit. If this issue needs to be revisited in the future, we could open a new PR.